packages/twenty-server/src/modules/messaging/message-import-manager/utils/is-sync-stale.util.ts

```
@@ -0,0 +1,17 @@
+import { MESSAGING_IMPORT_ONGOING_SYNC_TIMEOUT } from 'src/modules/messaging/message-import-manager/constants/messaging-import-ongoing-sync-timeout.constant';
+
+export const isSyncStale = (syncStageStartedAt: string): boolean => {
+  try {
+    const syncStageStartedTime = new Date(syncStageStartedAt).getTime();
+
+    if (isNaN(syncStageStartedTime)) {
+      throw new Error('Invalid date format');
+    }
+
+    return (
+      Date.now() - syncStageStartedTime > MESSAGING_IMPORT_ONGOING_SYNC_TIMEOUT
+    );
+  } catch (error) {
+    return false;
```

I would not catch it here, and let it throw. Here it could be silently failing